src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Logging/RoslynLogger.cs

```diff
@@ -139,7 +143,10 @@ public void LogBlockEnd(FunctionId functionId, LogMessage logMessage, int blockI
                 return;
             }
 
-            var properties = GetProperties(functionId, logMessage, delta);
+            using var pooledObject = s_propertyPool.GetPooledObject();
```

Comment:
@CyrusNajmabadi  -- I believe one thought you mentioned involved changing the LogMessage type passed into this method. I'd definitely prefer not making that level of change for this. I think just using a pooled List can give us the allocation free telemetry (at least from this level's perspective)